client/src/ui/BitbakeTerminal.ts

```diff
@@ -139,6 +153,10 @@ export class BitbakePseudoTerminal implements vscode.Pseudoterminal {
     processResolved.on('exit', (code) => {
       this.lastExitCode = code ?? -1
       this.process = undefined
+      if (this.terminalName.includes('Bitbake: Scan recipe env')) {
```

Comment:
The BitbakeTerminal class is about presenting bitbake processes, it should not handle communication with the server. Also it's not ideal to test through the terminalName. I think this should rather be done in the caller of the taskProvider (the one that will compose the task). I'm not sure you have added it yet to the PR?